lua/entities/ent_fedora_point.lua

```diff
@@ -21,7 +21,13 @@ end
 
 function ENT:Think()
 	local ply = self.Owner
-	if SERVER then return end
+	if SERVER then
+		if !IsValid(self:GetOwner()) then

```

Comment:
I meant GetActiveWeapon returns nil if the player has no weapon, so it doesn’t work in that case. You can also clean up the if statement like:
“If not isvalid(owner) or not isvalid(owner.weapon) or not owner.weapon.class==fedora then remove”
That works because of short circuit boolean evaluation - if owner is invalid, it won’t run the later statements so it won’t error

Reply:
yeah that was what I was afraid of, I didn't want to check if the owner was valid and if the owner's weapon was valid on the same statement because it would error if it checked later statements